src/Agent.Listener/Configuration/LinuxServiceControlManager.cs

```diff
@@ -100,37 +100,31 @@ public override void StartService(string serviceName)
                 ChangeOwnershipToLoginUser(filesToChange);
                 ReloadSystemd();
                 ExecuteSystemdCommand("start " + serviceName);
+                _term.WriteLine(StringUtil.Loc("ServiceStartedSuccessfully", serviceName));
             }
             catch (Exception)
             {
-                _term.WriteError(StringUtil.Loc("LinuxServiceStartFailed"));
+                _term.WriteError(StringUtil.Loc("CanNotStartService"));
                 throw;
             }
         }
 
-        protected virtual string GetUnitFile(string serviceName)
+        public override void StopService(string serviceName)
         {
-            return Path.Combine(SystemdPathPrefix, serviceName);
-        }
-
-        protected virtual bool CheckIfSystemdExists()
-        {
-            Trace.Info(nameof(CheckIfSystemdExists));
+            Trace.Info(nameof(StopService));
             try
             {
-                var commName = File.ReadAllText(InitFileCommandLocation).Trim();
-                return commName.Equals("systemd", StringComparison.OrdinalIgnoreCase);
+                ExecuteSystemdCommand("stop " + serviceName);
             }
-            catch (Exception ex)
+            catch (Exception)
```

Comment:
Done.. traced to file. This place we dont want to throw error on the terminal. We are trying to stop the linux service. Its ok if it fails. It may still work if we reconfigure the service